php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsCheck.java

```diff
@@ -35,43 +39,78 @@
 public class HardCodedCredentialsCheck extends PHPVisitorCheck {
 
   public static final String KEY = "S2068";
+  private static final String MESSAGE = "'%s' detected in this variable name, review this potentially hardcoded credential.";
+  private static final String DEFAULT_CREDENTIAL_WORDS = "password,passwd,pwd";
 
-  private static final String MESSAGE = "Remove this hard-coded password.";
+  @RuleProperty(
+    key = "credentialWords",
+    description = "Comma separated list of words identifying potential credentials",
+    defaultValue = DEFAULT_CREDENTIAL_WORDS)
+  public String credentialWords = DEFAULT_CREDENTIAL_WORDS;
 
-  private static final Pattern PASSWORD_LITERAL_PATTERN = Pattern.compile("password=..", Pattern.CASE_INSENSITIVE);
-  private static final Pattern PASSWORD_VARIABLE_PATTERN = Pattern.compile("password", Pattern.CASE_INSENSITIVE);
+  private List<Pattern> variablePatterns = null;
+  private Stream<Pattern> variablePatterns() {
+    if (variablePatterns == null) {
+      variablePatterns = toPatterns("");
+    }
+    return variablePatterns.stream();
+  }
+
+  private List<Pattern> literalPatterns = null;
```

Comment:
While I understand the idea behind this field placed here, I would still keep fields grouped at the top of the file.